aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/CredentialStoreActions.kt

```diff
@@ -66,11 +92,26 @@ object CredentialStoreActions : StoreActions {
             dispatcher.send(evt)
         }
 
-    override fun storeCredentialsAction(credentials: AmplifyCredential) =
+    override fun storeCredentialsAction(
+        amplifyCredentialType: AmplifyCredentialType,
+        username: String?,
+        credentials: AmplifyCredential
+    ) =
         Action<CredentialStoreEnvironment>("StoreCredentials") { id, dispatcher ->
             logger.verbose("$id Starting execution")
             val evt = try {
-                credentialStore.saveCredential(credentials)
+                when (amplifyCredentialType) {
+                    AmplifyCredentialType.AMPLIFY_CREDENTIAL -> {
+                        credentialStore.saveCredential(credentials)
+                    }
+                    AmplifyCredentialType.DEVICE_METADATA -> {
+                        credentialStore.saveDeviceMetadata(
+                            username as String,
```

Comment:
casting `String?` to `String` is code smell as when `username` is null, it will throw an NPE.
Please use null safe constructs such as `?.let`